subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1211,8 +1239,11 @@ int bt_bap_scan_delegator_add_src(const struct bt_bap_scan_delegator_add_src_par
 
 	internal_state = get_free_recv_state();
 	if (internal_state == NULL) {
-		LOG_DBG("Could not add src");
-
+		LOG_DBG("Could not add src. No free receive state.");
+		if (scan_delegator_cbs != NULL &&
+			scan_delegator_cbs->recv_states_full != NULL) {
+			scan_delegator_cbs->recv_states_full(BT_ATT_ERR_INSUFFICIENT_RESOURCES);
```

Comment:
If the callback uses `int` we should use errno values

Reply:
@Thalley Could you please elaborate on this comment? Thanks

Reply:
Sure: The value you are providing here, `BT_ATT_ERR_INSUFFICIENT_RESOURCES` is an ATT error code, which we usually don't expose directly to the application. Often our APIs just errno (like `-ENOMEM`). 

We do expose ATT error code for same cases, but then we usually document it as 
```
< 0  => errno
== 0 => success
> 0  => ATT error code
```